### PR TITLE
Append commit to tarball

### PR DESCRIPTION
**Pull request by abelboldu**

### CODE REVIEW

File: workflows/devnet-build/Jenkinsfile [hunk 1]
```diff
@@ -21,13 +22,16 @@ pipeline{
                     }
                     buildEnv.inside {
                         dir('lisk'){
-                            git url: 'https://github.com/LiskHQ/lisk', branch: "${params.branch}"
+                            checkout([$class: 'GitSCM', branches: [[name: "${params.branch}" ]],
```
Comment (ghost) [inline workflows/devnet-build/Jenkinsfile:25]: Why is this change necessary?

File: workflows/devnet-build/Jenkinsfile [hunk 2]
```diff
@@ -42,11 +46,14 @@ pipeline{
         }
         stage('Upload tarball'){
             steps {
-                dir("lisk-build") {
-                    sh """
-                    s3cmd put --acl-public release/lisk-${env.VERSION}-Linux-x86_64.tar.gz  s3:///lisk-releases/lisk-core/${params.tag}/lisk-${env.VERSION}-Linux-x86_64.tar.gz
-                    s3cmd put --acl-public release/lisk-${env.VERSION}-Linux-x86_64.tar.gz.SHA256 s3:///lisk-releases/lisk-core/${params.tag}/lisk-${env.VERSION}-Linux-x86_64.tar.gz.SHA256
-                    """
+                script {
+                    def outputFile = params.commitsha ? "lisk-${env.VERSION}-${env.COMMIT}-Linux-x86_64.tar.gz" : "lisk-${env.VERSION}-Linux-x86_64.tar.gz"
```
Comment (ghost) [inline workflows/devnet-build/Jenkinsfile:50]: Won't the `*.SHA256` contain the wrong file name?

File: workflows/devnet-build/Jenkinsfile (hunk 1 above)
Comment (abelboldu) [reply workflows/devnet-build/Jenkinsfile:25]: Because with the previous syntax you can specify a branch, but not a SHA or a tag.

File: workflows/devnet-build/Jenkinsfile (hunk 2 above)
Comment (abelboldu) [reply workflows/devnet-build/Jenkinsfile:50]: Yes you're right here... I will need to re-generate it

File: workflows/devnet-build/Jenkinsfile [hunk 3]
```diff
@@ -42,11 +46,18 @@ pipeline{
...
+                    dir("lisk-build") {
+                        sh """
+                        if [ ! -f release/${outputFile} ]; then
+                            mv release/lisk-${env.VERSION}-Linux-x86_64.tar.gz release/${outputFile}
+                            sha256sum release/${outputFile} > release/${outputFile}.SHA256
```
Comment (ghost) [inline workflows/devnet-build/Jenkinsfile]: That'll compute the sha256 checksum for `release/${outputFile}` instead of just `${outputFile}` which would be much nicer.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
